awx/main/scheduler/task_manager.py

```diff
@@ -518,6 +519,19 @@ def process_pending_tasks(self, pending_tasks):
             if not found_acceptable_queue:
                 logger.debug("{} couldn't be scheduled on graph, waiting for next cycle".format(task.log_format))
 
+    def timeout_approval_node(self):
+        workflow_approval = WorkflowApproval.objects.filter(status='pending').prefetch_related('workflow_approval_template')
+        now = tz_now()
+        for task in workflow_approval:
+            approval_timeout_seconds = timedelta(seconds=task.workflow_approval_template.timeout)
+            if task.workflow_approval_template.timeout == 0:
+                continue
+            if (now - task.created) >= approval_timeout_seconds:
+                logger.info("This approval node has timed out.")
```

Comment:
It'll help everyone out to log _which_ approval node timed out. A rudimentary way to format the primary key in the text.